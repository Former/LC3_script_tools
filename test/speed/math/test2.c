
int main()
{
    int k, i, res = 0;

    for (k = 0; k < 30; ++k)
    {
        for (i = 0; i < 15000; ++i)
        {
            res += (i >> 1 + k << 2) & 0xFF;
            res &= 0x1F;
        }
        printf("%d res = %d\n", k, res);
    }

    printf("Test finish %d\n", res);
    return 0;
}
