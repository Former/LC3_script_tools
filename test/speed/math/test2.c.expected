0 res = 8
1 res = 8
2 res = 8
3 res = 24
4 res = 8
5 res = 24
6 res = 8
7 res = 24
8 res = 24
9 res = 24
10 res = 24
11 res = 24
12 res = 24
13 res = 24
14 res = 24
15 res = 24
16 res = 24
17 res = 24
18 res = 24
19 res = 24
20 res = 24
21 res = 24
22 res = 24
23 res = 24
24 res = 24
25 res = 24
26 res = 24
27 res = 24
28 res = 24
29 res = 24
Test finish 24
